json/src/main/java/com/basistech/rosette/dm/jackson/TranslatedDataMixin.java

```diff
@@ -26,7 +26,10 @@
  */
 public abstract class TranslatedDataMixin {
     @JsonCreator
-    TranslatedDataMixin(@JsonProperty("domain") TextDomain domain,
+    TranslatedDataMixin(
+                        @JsonProperty("sourceDomain") TextDomain sourceDomain,
+                        @JsonProperty("targetDomain") TextDomain targetDomain,
```

Comment:
@Katsuya-Tomioka, is it backwards-compatible to modify this mixin constructor like this? Is it possible to rename a field (`domain` to `targetDomain`) with Jackson without deprecating the old field and making a new field?

Reply:
Yeah, I was spending some time here. I really didn't find a good way to have both fields. The only thing I can think of is to not rename it and just leave in `domain` for the target domain.  That's not really nice though :smile:

Reply:
`@JsonAlias` might just do the trick here

Reply:
I'd suggest not to rename the field "domain" or in this case accessor for the field, so that it won't break customer's code. Builders are fine; no one but us use that. But we do want to keep changes to the model additive. Can you add a deprecated "getDomain()"?

Reply:
So you don't agree with the current solution either? using `@JsonAlias`

```java
// TODO: remove @JsonAlias("domain") when TranslatedData.domain is removed
@JsonProperty("targetDomain") @JsonAlias("domain") TextDomain targetDomain,
```